src/App.tsx

```diff
@@ -1,26 +1,254 @@
+} from 'react';
+import { addTodo, changeTodo, deleteTodos, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterOptions } from './enums/FilterOptions';
+import { ErrorMessages } from './enums/ErrorMessages';
+import { filteredTodos } from './utils/filteringTodos';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { USER_ID } from './utils/USER_ID';
+import { Errors } from './components/Errors';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<FilterOptions>(FilterOptions.ALL);
+  const [errorMessage, setErrorMessage] = useState(ErrorMessages.NO_ERRORS);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [shouldFocus, setShouldFocus] = useState(true);
+  const [loadingTodosCount, setLoadingTodosCount] = useState<number[]>([]);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [todos]);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+  };
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (inputValue.trim() === '') {
+      setErrorMessage(ErrorMessages.EMPTY_TITLE);
+
+      return;
+    }
+
+    const temporaryTodo: Todo = {
+      id: 0,
+      title: inputValue.trim(),
+      userId: USER_ID,
+      completed: false,
+    };
+
+    setTempTodo(temporaryTodo);
+    setIsSubmitting(true);
+    setShouldFocus(false);
+    setLoadingTodosCount(currentCount => [...currentCount, 0]);
+
+    addTodo({ title: inputValue.trim(), userId: USER_ID, completed: false })
+      .then((createdTodo: Todo) => {
+        setTodos((currentTodos: Todo[]) => [...currentTodos, createdTodo]);
+        setInputValue('');
+        setErrorMessage(ErrorMessages.NO_ERRORS);
+        setShouldFocus(true);
+
+        if (inputRef.current) {
+          inputRef.current.focus();
+        }
+      })
+      .catch(() => {
+        setShouldFocus(true);
+        setErrorMessage(ErrorMessages.ADDING_ERROR);
+
+        if (inputRef.current) {
+          inputRef.current.focus();
+        }
+      })
+      .finally(() => {
+        if (inputRef.current) {
+          inputRef.current.disabled = false;
+          inputRef.current.focus();
+        }
+
+        setIsSubmitting(false);
+        setTempTodo(null);
+        setLoadingTodosCount(currentCount =>
+          currentCount.filter(todoId => todoId !== 0),
+        );
+      });
+  };
+
+  const handleTodoDelete = useCallback(
+    (todoId: number) => {
+      setIsSubmitting(true);
```

Comment:
`isSubmitting` its a loader that is displayed when you are manipulating with todos, maybe it's better to choose a more descriptive name for it